.github/bin/runtests.sh

```diff
@@ -0,0 +1,157 @@
+#    POSSIBILITY OF SUCH DAMAGES. SOME JURISDICTIONS DO NOT ALLOW THE EXCLUSION
+#    OR LIMITATION OF LIABILITY FOR PERSONAL INJURY, OR OF INCIDENTAL OR
+#    CONSEQUENTIAL DAMAGES, SO THIS LIMITATION MAY NOT APPLY TO YOU. In no event
+#    shall THE AUTHOR's total liability to you for all damages (other than as may
+#    be required by applicable law in cases involving personal injury) exceed
+#    the amount of five dollars ($5.00). The foregoing limitations will apply
+#    even if the above stated remedy fails of its essential purpose.
+################################################################################
+
+# bootstrap checks (pre pre-checks)
+EXIT_CODE=0
+
+test -x `command -v uname` || exit 1 ;
+test -x `command -v git` || exit 1 ;
+
+if [[ ( $(uname -s) = "Darwin" ) ]] ; then
+	export CI_OS_NAME="osx" ;
+else
+	export CI_OS_NAME="linux" ;
+fi
+
+
+# if locking was needed:
+####
+# ulimit -t 600
+# PATH="/bin:/sbin:/usr/sbin:/usr/bin"
+# umask 137
+#
+LOCK_FILE="/tmp/CI_test_script_lock"
+ 
+if [[ -f ${LOCK_FILE} ]] ; then
+	echo "Lock found! ABORT"
+ 	exit ${EXIT_CODE:-0} ;
+fi
+# 
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGABRT
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGHUP
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGTERM
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGQUIT
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGINT
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGKILL
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit ${EXIT_CODE:-0} ;' EXIT
+# 
+touch ${LOCK_FILE} 2>/dev/null || exit 0 ;
+
+# THIS IS THE ACTUAL WORK:
+
+export CI_TOOLS_BIN=$(dirname "${0}") ;
+#TODO: make this more absolute
+export CI_TEST_ROOT=$(dirname ${PWD:-./}/.) ;
+
+${CI_TOOLS_BIN}/pretests.sh
+
+function fn_test_pass() {
+	echo "... Passed" ;
+	return 0 ;
+}
+
+function fn_test_fail() {
+	echo "... FAILED" ;
+	return 1 ;
+} 1>&2
+
+function fn_test_cmd() {
+	local CI_TEST_CASE_RESULT=0 ;
+	local OLDUMASK=$(umask) ;
+	( ${CI_TOOLS_BIN%/}/${1} ${@:2:$#-1} ) || CI_TEST_CASE_RESULT=$? ;
+	wait ;
+	# clean up
+	umask "$OLDUMASK" ;
+	# return to start path
+	cd "${CI_TEST_ROOT}" ;
+	return $CI_TEST_CASE_RESULT ;
+}
+
+echo "TEST START" || EXIT_CODE=2 ;
+date ;
+
+echo "Example test for usage output" || EXIT_CODE=2 ;
+[[ ( $(fn_test_cmd ../../rsync_tmbackup.sh -h | grep -ci -oF "Usage:") -ge 1 ) ]] && fn_test_pass || fn_test_fail || EXIT_CODE=1 ;
+
+echo "Example fail test"
+# the || true prevents the test from erroring because the fail is expected here: use || EXIT_CODE=1 for expected pass
+[[ ( $( fn_test_cmd no such test ) ) ]] && fn_test_pass || fn_test_fail || true ;
+
+
+cd "${CI_TEST_ROOT}" || true ;
+
+##############################
+# can more run setup code here
+##############################
+
+
+echo "TEST END" ;
+
+### KEEP THIS LINE ###
+${CI_TOOLS_BIN}/posttests.sh
```

Comment:
This is where the cleanup script is called. caveat: it does nothing at the moment.